tests/test-files/test-dependencies.yaml

```diff
@@ -0,0 +1,21 @@
+version: 2.0
+steps:
+  step1:
+    run:
+      image: {{ DOCKER_REGISTRY }}/ubuntu:latest
+      cmd: echo "Hello from step 1"
+  step2:
+    depends:
+      - step3
+      - step4
+    run:
+      image: {{ DOCKER_REGISTRY }}/ubuntu:latest
+      cmd: echo "Hello from step 2"
+  step3:
+    run:
+      image: {{ DOCKER_REGISTRY }}/ubuntu:latest
+      cmd: echo "Hello from step 3"
+  step4:
+    run:
+      image: {{ DOCKER_REGISTRY }}/ubuntu:latest
+      cmd: echo "Hello from step 4"
```

Comment:
This will run with the other tests, correct? It might be helpful to have the actual step2 depend on something produced by step3 and step4, like an artifact or something?

Reply:
I am not sure I understand what you mean? Currently the `depends` list will only accept valid step names which must be processed before it. There is no logic in the code to accept anything other than step names. Can you expand on what you would like to see in this test?

Perhaps my understanding of the requirements was not correct.

Reply:
I was saying that maybe the commands themselves in the steps should actually use artifacts from the previous steps to ensure that things have been run in the correct order as well. i.e. step2 commands would look at `/artifacts/step3/<artifact>` and `/artifacts/step4/<artifact>` to ensure those steps were run. This artifact could be a blank file even (i.e. the step3 and step4 just touch a file and then include it in the artifacts).

Reply:
Oh ok, I see. And added layer of validation. That makes sense, I like that, I will add that in.

Reply:
I was not able to write files to this location with buildrunner. I also tried writing to current dir which is /source (I think) and it also failed.

```
[16:11:28] [step3] cmd> echo "Hello from step 3" > /artifacts/step3/step3.txt
[16:11:28] [step3] /bin/sh: 1: cannot create /artifacts/step3/step3.txt: Read-only file system
```

Reply:
done